protocol/contracts/admin/TempleTeamPaymentsV2.sol

```diff
@@ -0,0 +1,84 @@
+//SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract TempleTeamPaymentsV2 is Ownable {
+    IERC20 public immutable temple;
+    uint256 public immutable roundStartDate;
+
+    mapping(address => uint256) public allocation;
+    mapping(address => uint256) public claimed;
+    mapping(address => bool) public paused;
+
+    event Claimed(address indexed member, uint256 amount);
+
+    constructor(IERC20 _temple, uint256 _roundStartDate) {
+        temple = _temple;
+        roundStartDate = _roundStartDate;
+    }
+
+    function setAllocations(
+        address[] memory _addresses,
+        uint256[] memory _amounts
+    ) external onlyOwner {
+        require(
+            _addresses.length == _amounts.length,
+            "TempleTeamPayments: addresses and amounts must be the same length"
+        );
+        address addressZero = address(0);
+        for (uint256 i = 0; i < _addresses.length; i++) {
+            require(
+                _addresses[i] != addressZero,
+                "TempleTeamPayments: Address cannot be 0x0"
+            );
+            allocation[_addresses[i]] = _amounts[i];
+        }
+    }
+
+    function setAllocation(
+        address _address,
+        uint256 _amount
+    ) external onlyOwner {
+        require(
+            _address != address(0),
+            "TempleTeamPayments: Address cannot be 0x0"
+        );
+        allocation[_address] = _amount;
+    }
+
+    function toggleMember(address _address) external onlyOwner {
+        paused[_address] = !paused[_address];
+    }
+
+    function withdrawToken(IERC20 _token, uint256 _amount) external onlyOwner {
+        require(
+            _amount > 0,
+            "TempleTeamPayments: Amount must be greater than 0"
+        );
+        SafeERC20.safeTransfer(_token, msg.sender, _amount);
+    }
+
+    function calculateClaimable(address _member) public view returns (uint256) {
```

Comment:
if there's no vesting, you only need to check if `claimed[address] == true` and use a `mapping(address => boolean)` instead.
also, if `setAllocation` is called for a user who already claimed but wrong allocation was set, you should reset claimed and override allocation.

Reply:
Initially the reason i kept the claimed mapping with uints was so the contract could then make it easier to track the total claimed by a user when the allocation is changed after claiming

For example a user may initially be allocated 1 token and call `claim()`
after that, `allocation[msg.sender] = 1e18` and `claimed[msg.sender] = 1e18`
Following that claim, the owner may decide to call `setAllocation` to allow them to claim 2 additional tokens from the funds
If the user calls claim again, now `allocation[msg.sender] = 2e18` and `claimed[msg.sender] = 3e18`

With the uints it becomes harder to check the total claimed, but its probably rare so if you would prefer I can go ahead and change it

Reply:
since allocations are mutable I think uint mapping makes sense